start_jupyter_cm/gnome.py

```diff
@@ -45,13 +45,15 @@ def add_jupyter_here():
              'notebook': os.path.join(logo_path, 'jupyter.png'),
              'lab': os.path.join(logo_path, 'jupyter.png')}
 
+    path_python,filename = os.path.split(shutil.which("python"))
```

Comment:
Would the following be more simple
```python
python_exec = shutil.which("python")
```
and adjust accordingly the shebang line? Some system python may be using slightly different name for specific python version, etc.

Reply:
Your suggested code is certainly cleaner than mine - my one thought in writing the code in the way I have is to make it clear in the script prototype what language is being used, so that future developers wouldn't have to check the code within the functions to determine this.

I agree that the system python *should* be fine. The only reason I thought about adding this fix was that my fresh install of ubuntu only has a "python3" executable - an attempt to avoid people running python 2 scripts in an incompatible environment I guess. But this then breaks any attempt to use which (both bash and python shutil versions) to find the executable, as this doesn't allow wildcards in the executable name.

Maybe when python 3 becomes the de facto standard then we can use the system python again, but while we're still in the cross-over period I don't think that will always work.

Reply:
I am not sure if I understand everything you mentioned, but yes, I definitely agree this is better to use the same python and this is a bug that we haven't issue with so far!

Do you agree with my suggested change?

Reply:
Yep - I'm happy with your suggested change & have included it in the code now.